packages/@aws-cdk/shared-interfaces/lib/asset-source.ts

```diff
@@ -0,0 +1,28 @@
+/**
+ * The minimum necessary fields to create an asset
+ */
+export interface AssetSource {
```

Comment:
```suggestion
export interface IAssetSource {
```

Otherwise it cannot be implemented by classes (jsii restriction)

Reply:
Really? It's been working so far when I implement it in the awscdk-asset-awscli repo. 

But good to know, I removed the `I` because I thought this was not a "behavioral" interface, because it only holds data. But now realizing an interface that would implemented by a class is still behavioral, because the class will end up doing stuff.